src/modules/pos/validator/components/ChangeCommission/Form/hooks/useCommissionChangeDate.js

```diff
@@ -0,0 +1,36 @@
+import { usePosConstants } from '@pos/validator/hooks/queries';
+import { useCurrentAccount } from '@account/hooks';
+import { useTransactions } from '@transaction/hooks/queries';
+import { MODULE_COMMANDS_NAME_MAP } from '@transaction/configuration/moduleCommand';
+import { useNetworkStatus } from '@network/hooks/queries';
+
+export const useCommissionChangeDate = () => {
+  const { data: posConstants } = usePosConstants();
+  const [
+    {
+      metadata: { address },
+    },
+  ] = useCurrentAccount();
+  const { data: transactions } = useTransactions({
+    config: {
+      params: {
+        address,
+        moduleCommand: MODULE_COMMANDS_NAME_MAP.changeCommission,
+        limit: 1,
+        sort: 'timestamp:desc',
+      },
+    },
+  });
+  const lastChangeCommissionTimestamp = transactions?.data[0]?.block?.timestamp;
+  const networkStatus = useNetworkStatus();
+
+  return (
+    lastChangeCommissionTimestamp &&
+    new Date(
+      (lastChangeCommissionTimestamp +
+        posConstants.data?.commissionIncreasePeriod *
+          networkStatus.data?.data?.genesis?.blockTime) *
+        1000
+    )
```

Comment:
to make this easier to read, I think its best to store this in a variable(with a good naming) possibly on line 26 then consume here

Reply:
I rather only do the calculation and time creation when it is needed (if `lastChangeCommissionTimestamp` is true). Refactored that code to use a function though, in this https://github.com/LiskHQ/lisk-desktop/pull/5092/commits/3f1c2892ddaa049520e1f0a0f653a5a73c764d81